File: backend/app/models/country.py
"""
Country Model.

Maps to TR_COU_Country table in the database.
"""
from sqlalchemy import Column, Integer, String
from app.models.base import Base


class Country(Base):
    """Country model for TR_COU_Country table."""
    
    __tablename__ = "TR_COU_Country"
    
    cou_id = Column(Integer, primary_key=True, autoincrement=True)
    cou_name = Column(String(200), nullable=False)
    cou_code = Column(String(50), nullable=True)
    cou_iso_code = Column(String(50), nullable=True)
    
    # Aliases for consistent naming
    @property
    def id(self):
        return self.cou_id
    
    @property
    def name(self):
        return self.cou_name
    
    @property
    def code(self):
        return self.cou_code
    
    # Alias for frontend compatibility (uses ctr_ prefix)
    @property
    def ctr_id(self):
        return self.cou_id
    
    @property
    def ctr_name(self):
        return self.cou_name
    
    def __repr__(self):
        return f"<Country(id={self.cou_id}, name='{self.cou_name}')>"

File: Legacy/ERP.Entities/Country.cs
using System;

namespace ERP.Entities
{
    /// <summary>
    /// Country entity model for TR_COU_Country table
    /// Used for geographic reference data
    /// </summary>
    [Serializable]
    public class Country : BaseClass
    {
        /// <summary>
        /// Country ID (cou_id)
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Country name (cou_name)
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Country code (cou_code) - e.g., "FR", "DE"
        /// </summary>
        public string Code { get; set; }

        /// <summary>
        /// ISO country code (cou_iso_code) - e.g., "FRA", "DEU"
        /// </summary>
        public string IsoCode { get; set; }
    }
}
